llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -1230,6 +1230,10 @@ static Value *simplifySDivInst(Value *Op0, Value *Op1, bool IsExact,
   // If two operands are negated and no signed overflow, return -1.
   if (isKnownNegation(Op0, Op1, /*NeedNSW=*/true))
     return Constant::getAllOnesValue(Op0->getType());
+  Value *X, *Y;
```

Comment:
It has been handled by `isKnownNegation`:
https://github.com/llvm/llvm-project/blob/a604c4b562e99470e397f050f1e8707f923ebed7/llvm/lib/Analysis/ValueTracking.cpp#L7610-L7615

I think the motivation of #69574 is to prove that `(a - b) / (b - a)` is non-zero.

Reply:
Clang failed to fold `(-a + b) / (a - b)` into `-1` due to the lack of `nsw` flag.
Alive2: https://alive2.llvm.org/ce/z/tPAt3K

You can create a separate patch to address this :)

Reply:
created https://github.com/llvm/llvm-project/pull/72127